### PR TITLE
Alerting docs: Update `Alert rules` intro and `Introduction` pages

### PR DESCRIPTION
**Pull request by ppcano**
Update [Alert rules](https://grafana.com/docs/grafana/next/alerting/fundamentals/alert-rules/) and [Introduction](https://grafana.com/docs/grafana/next/alerting/fundamentals/)

- Restructure a bit the sections and concepts to follow the Alert creation and handling process

- Highlight more the key features, differences, and concepts.

- Consistency using the `Alert instance` term. 

- Update diagrams.

### CODE REVIEW

File: docs/sources/alerting/fundamentals/_index.md [hunk 1]
```diff
@@ -35,72 +35,60 @@ The following concepts are key to your understanding of how Grafana Alerting wor
...
 
-Alerts are matched to notification policies using label matchers. These are human-readable expressions that assert if the alert's labels exactly match, do not exactly match, contain, or do not contain some expected text. For example, the matcher `foo=bar` matches alerts with the label `foo=bar` while the matcher `foo=~[a-zA-Z]+` matches alerts with any label called foo with a value that matches the regular expression `[a-zA-Z]+`.
+{{< figure src="/static/img/docs/alerting/unified/multi-dimensional-alert.png" caption="Multiple alert instances from a single alert rule" >}}
 
-By default, an alert can only match one notification policy. However, with the `continue` feature alerts can be made to match any number of notification policies at the same time. For more information on notification policies, see [fundamentals of Notification Policies][notification-policies].
+[Alert rules are frequently evaluated][alert-rule-evaluation] and updates the state of their alert instances accordingly. Only alert instances that are in a firing or resolved state are routed to notification policies to be handled.
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/_index.md:58]:
```suggestion
[Alert rules are frequently evaluated][alert-rule-evaluation] and the state of their alert instances is updated accordingly. Only alert instances that are in a firing or resolved state are routed to notification policies to be handled.
```

File: docs/sources/alerting/fundamentals/_index.md [hunk 2]
```diff
@@ -35,72 +35,60 @@ The following concepts are key to your understanding of how Grafana Alerting wor
...
+[Notification policies][notification-policies] group alerts and then route them to contact points. They determine when notifications are sent, and how often notifications should be repeated.
 
-## Provisioning
+Alert instances are matched to notification policies using label matchers. This provides a flexible way to organize and route alerts to different receivers.
 
-You can create your alerting resources (alert rules, notification policies, and so on) in the Grafana UI; configmaps, files and configuration management systems using file-based provisioning; and in Terraform using API-based provisioning.
+Each policy consists of a set of label matchers (0 or more) that specify which alert instances (identified by their labels) they are or aren’t interested in handling. Notification policies are defined as a tree structure where the root of the notification policy tree is called the Default notification policy, and each policy can have child policies.
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/_index.md:66]:
```suggestion
Each policy consists of a set of label matchers (0 or more) that specify which alert instances (identified by their labels) they handle. Notification policies are defined as a tree structure where the root of the notification policy tree is called the **Default notification policy**. Each policy can have child policies.
```

File: docs/sources/alerting/fundamentals/_index.md [hunk 3]
```diff
@@ -35,72 +35,60 @@ The following concepts are key to your understanding of how Grafana Alerting wor
...
 
-**Silence alerts**
+Grafana Alerting is built on the Prometheus model of designing alerting systems.
 
-Silences stop notifications from getting created and last for only a specified window of time.
-Silences allow you to stop receiving persistent notifications from one or more alert rules. You can also partially pause an alert based on certain criteria. Silences have their own dedicated section for better organization and visibility, so that you can scan your paused alert rules without cluttering the main alerting view.
+Prometheus-based alerting systems have two main components:
 
-**Mute timings**
+- an alert generator that evaluates alert rules and sends firing and resolved alerts to the alert receiver.
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/_index.md:86]:
```suggestion
- An alert generator that evaluates alert rules and sends firing and resolved alerts to the alert receiver.
```

File: docs/sources/alerting/fundamentals/_index.md [hunk 4]
```diff
@@ -35,72 +35,60 @@ The following concepts are key to your understanding of how Grafana Alerting wor
...
+- an alert receiver (also known as Alertmanager) that receives the alerts and is responsible for handling them and sending their notifications.
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/_index.md:87]:
```suggestion
- An alert receiver (also known as Alertmanager) that receives the alerts and is responsible for handling them and sending their notifications.
```

File: docs/sources/alerting/fundamentals/_index.md [hunk 5]
```diff
@@ -35,72 +35,60 @@ The following concepts are key to your understanding of how Grafana Alerting wor
...
 
-A mute timing is a recurring interval of time when no new notifications for a policy are generated or sent. Use them to prevent alerts from firing a specific and reoccurring period, for example, a regular maintenance period.
+Grafana doesn’t use Prometheus as its default alert generator because Grafana Alerting needs to work with many other data sources in addition to Prometheus.
 
-Similar to silences, mute timings do not prevent alert rules from being evaluated, nor do they stop alert instances from being shown in the user interface. They only prevent notifications from being created.
+However, Grafana can also use Prometheus as alert generator and external alertmanagers. For more information about how to use distinct alerting systems, refer to the [Grafana alert rule types][alert-rules].
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/_index.md:91]:
```suggestion
However, Grafana can also use Prometheus as an alert generator as well as external Alertmanagers. For more information about how to use distinct alerting systems, refer to the [Grafana alert rule types][alert-rules].
```

File: docs/sources/alerting/fundamentals/alert-rules/_index.md [hunk 6]
```diff
@@ -21,31 +21,28 @@ weight: 100
...
 # Alert rules
 
-An alert rule is a set of evaluation criteria for when an alert rule should fire. An alert rule consists of one or more [queries and expressions, a condition][queries-and-conditions], and the duration over which the condition needs to be met to start firing.
+An alert rule is a set of evaluation criteria for when an alert rule should fire. An alert rule consists of:
 
-While queries and expressions select the data set to evaluate, a condition sets the threshold that an alert must meet or exceed to create an alert.
+- queries and expressions that select the data set to evaluate.
+- conditions (the threshold) that the query must meet or exceed to trigger the alert instance.
+- an interval that specifies the frequency of [alert rule evaluation][alert-rule-evaluation] and a duration indicating how long the condition must be met to trigger the alert instance.
+- other options like the behavior in the absence of data, notification messages, and more.
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/alert-rules/_index.md:30]:
```suggestion
- Queries and expressions that select the data set to evaluate.
- A condition (the threshold) that the query must meet or exceed to trigger the alert instance.
- An interval that specifies the frequency of [alert rule evaluation][alert-rule-evaluation] and a duration indicating how long the condition must be met to trigger the alert instance.
- Other customizable options, for example, setting what should happen in the absence of data, notification messages, and more.

```

File: docs/sources/alerting/fundamentals/alert-rules/_index.md [hunk 7]
```diff
@@ -21,31 +21,28 @@ weight: 100
...
 
-An interval specifies how frequently an [alert rule is evaluated][alert-rule-evaluation]. Duration, when configured, indicates how long a condition must be met. The alert rules can also define alerting behavior in the absence of data.
-
-Grafana supports two different alert rule types: [Grafana-managed alert rules](#grafana-managed-alert-rules) and [Data source-managed alert rules](#data-source-managed-alert-rules).
+Grafana supports two different alert rule types: Grafana-managed alert rules and Data source-managed alert rules.
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/alert-rules/_index.md:31]:
```suggestion
Grafana supports two different alert rule types: Grafana-managed alert rules and data source-managed alert rules.
```

File: docs/sources/alerting/fundamentals/notifications/alertmanager.md [hunk 8]
```diff
@@ -18,7 +18,7 @@ weight: 111
 
 Grafana sends firing and resolved alerts to Alertmanagers. The Alertmanager receives alerts, handles silencing, inhibition, grouping, and routing by sending notifications out via your channel of choice, for example, email or Slack.
 
-Grafana has its own Alertmanager, referred to as "Grafana" in the user interface, but also supports sending alerts to other Alertmanagers too, such as the [Prometheus Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/).
+Grafana has its own Alertmanager, referred to as "Grafana" in the user interface, but also supports sending alerts to other Alertmanagers, such as the [Prometheus Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/). You can use both internal and external alertmanagers.
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/notifications/alertmanager.md:21]:
```suggestion
Grafana has its own Alertmanager, referred to as "Grafana" in the user interface, but also supports sending alerts to other Alertmanagers, such as the [Prometheus Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/). You can use both internal and external Alertmanagers.
```

**Comment by brendamuir** [comment]
lgtm-thank you. the only thing is im not sure i like "design your alerting system" still at the bottom of the intro - but that can be tackled later!

**Comment by ppcano** [comment]
> the only thing is im not sure i like "design your alerting system" still at the bottom of the intro
Yes, also, i think there is room for improvement in that section. 

> but that can be tackled later!
Agree.  I wanted to focus on the concepts first. 

@brendamuir I have added all your suggestions. Thanks for the back and forth and the reviews! Looking for approval to merge this.

**Review by brendamuir — APPROVED** [review]

**Pull request merged**